### PR TITLE
[FLINK-35868][cdc-connector][mongodb] Bump dependency version to support MongoDB 7.0

### PR DESCRIPTION
**Pull request by yuxiqian**
This closes FLINK-35868.

By bumping mongo-kafka and mongo-driver dependencies, Mongo CDC should be able to cooperate with MongoDB 6.0.16 & 7.0.12 (and legacy 5.0.2).

Notice: since JUnit doesn't allow parameterized `@ClassRule` or static fields, so containers must be created / destroyed before / after each test case, which significantly slows down the. testing (~50 min.). More discussion is required on this.

### CODE REVIEW

**Comment by yuxiqian** [comment]
@leonardBang @Jiabao-Sun PTAL

**Review by Jiabao-Sun — APPROVED** [review]
Thanks @yuxiqian for this improvement.
LGTM.

**Comment by Jiabao-Sun** [comment]
Shall we bump the driver version from 4.7.1 to 5.1.1 as well?

**Comment by yuxiqian** [comment]
> Shall we bump the driver version from 4.7.1 to 5.1.1 as well?

Done, bumped `mongo-kafka` version, too.

File: flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mongodb-cdc/pom.xml [hunk 1]
```diff
@@ -69,7 +69,7 @@ limitations under the License.
         <dependency>
             <groupId>org.mongodb</groupId>
             <artifactId>mongodb-driver-sync</artifactId>
-            <version>4.9.1</version>
+            <version>5.1.1</version>
```
Comment (leonardBang) [inline flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mongodb-cdc/pom.xml:72]: Could you also update the docs ？ and the older(release-3.1) version docs seem use an incorrect driver version.
https://nightlies.apache.org/flink/flink-cdc-docs-release-3.1/docs/connectors/flink-sources/overview/

**Review by Jiabao-Sun — CHANGES_REQUESTED** [review]
The CI failure should be related to the `mongo-kafka` version upgrade. 
We need to analyze the cause.

**Comment by yuxiqian** [comment]
I'll investigate this.

**Comment by yuxiqian** [comment]
Seems this commit https://github.com/mongodb/mongo-java-driver/commit/a03649cb24456f53db45e3a332c99d40a124748e changed return type of `listCollectionNames` and could not work properly with `mongo-kafka-connect`. Seems it's not 100% safe to use `5.9.1` too, since `mongo-kafka` still sticks to `5.7.x`.

Any thoughts from @Jiabao-Sun?

**Comment by Jiabao-Sun** [comment]
Thanks @yuxiqian for the investigation.
How about bump the driver version to 4.11?

File: docs/content.zh/docs/connectors/flink-sources/overview.md [hunk 2]
```diff
@@ -37,7 +37,7 @@ You can also read [tutorials]({{< ref "docs/connectors/flink-sources/tutorials/b
...
-| [mongodb-cdc]({{< ref "docs/connectors/flink-sources/mongodb-cdc" >}})     | <li> [MongoDB](https://www.mongodb.com): 3.6, 4.x, 5.0                                                                                                                                                                                                                                                                                                                                                 | MongoDB Driver: 4.3.4     |
+| [mongodb-cdc]({{< ref "docs/connectors/flink-sources/mongodb-cdc" >}})     | <li> [MongoDB](https://www.mongodb.com): 3.6, 4.x, 5.0                                                                                                                                                                                                                                                                                                                                                 | MongoDB Driver: 4.11.2    |
```
Comment (Jiabao-Sun) [inline docs/content.zh/docs/connectors/flink-sources/overview.md:40]: We can add 6.0, 6.1, 7.0 versions now.

Comment (yuxiqian) [reply docs/content.zh/docs/connectors/flink-sources/overview.md]: Done :)

**Review by Jiabao-Sun — APPROVED** [review]
Thanks @yuxiqian for this great work. LGTM now. 
cc @leonardBang

**Review by leonardBang — APPROVED** [review]
+1， Thanks @yuxiqian and @Jiabao-Sun for the nice work!

**Pull request merged**